docker/observability/prometheus/rules.yml

```diff
@@ -0,0 +1,111 @@
+            will be terminated, causing interruption of service.
+
+            This will happen if you deploy new PgBouncer pods and fail to roll
+            the application deployments that use session pooled connections
+            through them. If you haven't yet rolled services like
+            payments-service, this should be your first action.
+
+            Tail the pod logs to check we're waiting on outstanding connections
+            and confirm which database is affected by logging into the pod and
+            querying the PgBouncer admin console.
+
+            ```
+            $ kubectl logs -n {{ $labels.namespace }} {{ $labels.pod_name }} | grep outstanding_connections | tail -n 10
+            component=pgbouncer.child event=outstanding_connections database=postgres count=6
+            $ kubectl -n {{ $labels.namespace }} exec -it {{ $labels.pod_name }} bash
+            postgres:/$ psql -p 6432 -U pgbouncer pgbouncer -h /tmp
+            psql (11.2 (Ubuntu 11.2-1.pgdg18.04+1), server 1.9.0/bouncer)
+            Type "help" for help.
+
+            pgbouncer=# show databases;
+            ...
+            ```
+      - alert: StolonPgBouncerStaleReload
+        # Ensure we don't alert when shutdown_seconds > 0, as we expect not to
+        # reload PgBouncer when we're shutting down. Pending shutdowns should be
+        # caught by the StolonPgBouncerPendingShutdown alert, not this.
+        expr: >
+          max by (pod_name, namespace, version) (
+            stolon_pgbouncer_cluster_identifier * ignoring(cluster_name, store_prefix) (
+              (time() - stolon_pgbouncer_last_reload_seconds) > (2 * stolon_pgbouncer_store_poll_interval and stolon_pgbouncer_shutdown_seconds == 0)
+            )
+          )
+        labels:
+          severity: critical
+        annotations:
+          summary: Failed to reload PgBouncer on {{ $labels.pod_name }}
+          description: |
+
+            stolon-pgbouncer tries to reload PgBouncer every store poll interval
+            seconds, but this pod has failed to do so for more than 2x this
+            interval.
+
+            Either our connection to etcd is broken or PgBouncer is
+            unresponsive, either of which mean database failover won't update
+            this PgBouncer.
+
+            Logging into the pod and checking the live processes can show
+            whether a PgBouncer has died, and accessing the console might help
+            determine the route of the problem.
+
+            ```
+            $ kubectl -n {{ $labels.namespace }} exec -it {{ $labels.pod_name }} bash
+            postgres:/$ ps auxf
+            ... # expect to see PgBouncer and stolon-pgbouncer
+            postgres:/$ psql -p 6432 -U pgbouncer pgbouncer -h /tmp
+            psql (11.2 (Ubuntu 11.2-1.pgdg18.04+1), server 1.9.0/bouncer)
+            Type "help" for help.
+
+            pgbouncer=# show databases;
+            ...
+            ```
```

Comment:
Is this duplicated with rules that we have internally at GC?

Reply:
It is slightly yeah, minus some things like our dashboard link. Not sure what the best way to go around this is though, definitely worth having them in this repo for local testing but quite difficult to sync up.